cmd/prometheus/main.go

```diff
@@ -46,6 +46,24 @@ func main() {
 	os.Exit(Main())
 }
 
+var (
+	configAttempts = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "prometheus",
+		Name:      "configuration_attempts",
+		Help:      "number of times a configuration reload was attempted.",
+	})
+	configSuccesses = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "prometheus",
+		Name:      "configuration_successes",
+		Help:      "number of times a configuration reload was successful.",
+	})
+	configSuccessTime = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "prometheus",
+		Name:      "configuration_last_success",
+		Help:      "Timestamp of the last successful configuration reload.",
+	})
+)
+
```

Comment:
:+1: for that, though the latter would be out of scope for this PR :)

Do we even want to keep the last success timestamp metric then? It feels like the `config_last_reload_successful` metric would be more useful than the last successful timestamp in the use cases I can think of right now.